libs/langchain/langchain/embeddings/vertexai.py

```diff
@@ -36,11 +40,32 @@ def embed_documents(
         Returns:
             List of embeddings, one for each text.
         """
+        if self.show_progress_bar:
+            try:
+                from tqdm import tqdm
+            except ImportError:
+                logger.warning(
+                    "Unable to show progress bar because tqdm could not be imported. "
+                    "Please install with `pip install tqdm`."
+                )
+                progress_bar = None
+            else:
+                progress_bar = tqdm(total=len(texts), desc="VertexAIEmbeddings")
+        else:
+            progress_bar = None
```

Comment:
I think theres a clear way to do this where you dont have use the progress-bar update and close below
```suggestion
        if self.show_progress_bar:
            try:
                from tqdm import tqdm
            except ImportError:
                logger.warning(
                    "Unable to show progress bar because tqdm could not be imported. "
                    "Please install with `pip install tqdm`."
                )
                _iterator = range(0, len(texts), batch_size)
            else:
                _iterator = tqdm(range(0, len(texts), batch_size), total=len(texts), desc="VertexAIEmbeddings")
        else:
            _iterator = range(0, len(texts), batch_size)
```

Reply:
@hwchase17 
I'm using the progress-bar update so people know the total number of docs. To me, maybe just me, that's important. The tricky part is that we are using a batch size, so that's why I'm manually updating the progress bar.

In your approach, you are creating a `range(0, len(texts), batch_size)` but then setting the total to `len(texts)`. With that the progress bar will increase 1 by 1, but internally it's progressing at batch size, which will make the progress bar not get filled completely.

Here you have a python script that you can execute on your shell to prove my point:

```python
from tqdm import tqdm
import time

texts = [1,2,3,4,5,6,7,8,9,10]
batch_size = 2
_iterator = tqdm(range(0, len(texts), batch_size), total=len(texts), desc="VertexAIEmbeddings")

for a in _iterator:
    time.sleep(1)
```
VertexAIEmbeddings:  50%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▌                                                                                                                    | 5/10 [00:06<00:06,  1.30s/it]